File: app/src/main/java/ru/ulstu_team/ulstuschedule/injection/module/ApplicationModule.java
package ru.ulstu_team.ulstuschedule.injection.module;

import android.app.Application;
import android.content.Context;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import ru.ulstu_team.ulstuschedule.data.JsonDownloadService;
import ru.ulstu_team.ulstuschedule.data.local.PrefsManager;
import ru.ulstu_team.ulstuschedule.injection.ApplicationContext;


/**
 * Provide application-level dependencies.
 */
@Module
public class ApplicationModule {

    protected final Application mApplication;

    public ApplicationModule(Application application) {
        mApplication = application;
    }

    @Provides
    Application provideApplication() {
        return mApplication;
    }

    @Provides
    @ApplicationContext
    Context provideContext() {
        return mApplication;
    }

    @Provides
    @Singleton
    JsonDownloadService provideJsonDownloaderService() {
        return new JsonDownloadService();
    }

    @Provides
    @Singleton
    PrefsManager providePrefsManager(@ApplicationContext Context context) { return new PrefsManager(context); }
}

File: app/src/main/java/ru/ulstu_team/ulstuschedule/injection/module/ApplicationModule.kt
package ru.ulstu_team.ulstuschedule.injection.module

import android.app.Application
import android.content.Context

import javax.inject.Singleton

import dagger.Module
import dagger.Provides
import ru.ulstu_team.ulstuschedule.data.local.PrefsManager
import ru.ulstu_team.ulstuschedule.data.remote.VolleySingleton
import ru.ulstu_team.ulstuschedule.injection.ApplicationContext


/**
 * Provide application-level dependencies.
 */
@Module
class ApplicationModule(private val mApplication: Application) {

    @Provides
    fun provideApplication(): Application {
        return mApplication
    }

    @Provides
    @ApplicationContext
    fun provideContext(): Context {
        return mApplication
    }

    @Provides
    @Singleton
    fun providePrefsManager(@ApplicationContext context: Context): PrefsManager {
        return PrefsManager(context)
    }

    @Provides
    @Singleton
    fun provideVolleySingleton(@ApplicationContext context: Context): VolleySingleton {
        return VolleySingleton.getInstance(context)
    }
}
